subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/MavenResolver.java

```diff
@@ -285,7 +285,8 @@ private static boolean isUniqueSnapshot(ModuleComponentIdentifier id) {
     }
 
     protected static boolean isNonUniqueSnapshot(ModuleComponentIdentifier moduleComponentIdentifier) {
-        return moduleComponentIdentifier.getVersion().endsWith("-SNAPSHOT");
+        return "SNAPSHOT".equals(moduleComponentIdentifier.getVersion()) ||
```

Comment:
:x: The test is redundant. If `equals` work, then `endsWith` will too. We should avoid too many tests because this code is very sensitive to performance.

Reply:
Hi @melix! Sorry, it was a silly mistake... I will submit an update.

My previous exchange with @floscher resulted in the agreement that allowing only literal `SNAPSHOT` or prefix `-SNAPSHOT` should suffice, forgoing some unusual scenarios and resulting in simpler implementation.

Indeed, performance was a major concern when adding the `equals` check, against using a regular expression instead.